lib/_http_outgoing.js

```diff
@@ -424,6 +427,10 @@ function _storeHeader(firstLine, headers) {
         (state.contLen || this.useChunkedEncodingByDefault || this.agent);
     if (shouldSendKeepAlive) {
       header += 'Connection: keep-alive\r\n';
+      if (this._keepAliveTimeout) {
+        const timeoutSeconds = MathFloor(this._keepAliveTimeout) / 1000;
```

Comment:
Should we always set this to less than the actual timeout? And if so what is a good algorithm?

Reply:
Based https://github.com/mcollina/undici/pull/279 I feel that any client should always assume a value **less** than the hint. So further reducing it here can be counter productive.